src/SDK/Trace/Behavior/SpanExporterTrait.php

```diff
@@ -33,15 +39,59 @@ abstract public static function fromConnectionString(string $endpointUrl, string
      *
      * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.7.0/specification/trace/sdk.md#exportbatch
      *
+     * @psalm-suppress ArgumentTypeCoercion
      * @psalm-return SpanExporterInterface::STATUS_*
      */
     public function export(iterable $spans, ?CancellationInterface $cancellation = null): int
     {
+        $shouldRetry = true;
+        $status = SpanExporterInterface::STATUS_SUCCESS;
+        $attempt = 0;
         if (!$this->running) {
             return SpanExporterInterface::STATUS_FAILED_NOT_RETRYABLE;
         }
 
-        return $this->doExport($spans); /** @phpstan-ignore-line */
+        // If retryPolicy is not set then just doExport once and return the status
+        if ($this->retryPolicy === null) {
+            self::logDebug('retry polict is null. Exporting the spans without retry');
+
+            return $this->doExport($spans); /** @phpstan-ignore-line */
+        }
+
+        while ($shouldRetry) {
+            // find the delay time before retry (it is 0 for the first attempt)
+            $delay = $this->retryPolicy->getDelay($attempt);
```

Comment:
Having the retry logic in the trait instead of the exporter complicates handling `Retry-After` header & co.
[OTLP/HTTP Throttling](https://github.com/open-telemetry/opentelemetry-specification/blob/dbea54d2a416fe9e487d78dd33f8ab9251596bc2/specification/protocol/otlp.md#otlphttp-throttling)

Reply:
I have to put the retry logic in trait only because it is applicable to all span exporters and export() is actually implemented in that only.

Reply:
@Nevay : Can you please give me an example where I can implement this?

Reply:
I think we can ignore this; exporters that have to handle `Retry-After` should implement their own retry logic instead of using this trait.

Reply:
ok I will move the logic to exporter